dags/build_export_dag.py

```diff
@@ -112,7 +119,43 @@ def copy_from_export_path(export_path, file_path):
         else:
             download_from_gcs(bucket=output_bucket, object=export_path + filename, filename=file_path)
 
-    def get_block_range(tempdir, date):
+
+    def node_watch_command(**kwargs):
+        """
+        This functions checks if the provider_uri is connected or not.
+        input: kwargs
+        returns: xcom_push either provider_uri or one of the uris from backup_provider_uri_list 
+        to the dependencies. 
+        Xcoms table in database needs to be cleaned up time by time
+        """
+        def connection_check_recursice(uri,backup_uri_list):
```

Comment:
I like that you used backup_uri_list instead of just backup_uri. I think for simplicity it can be just collapsed to provider_uris which includes both the primary uri and the backup uris.

One issue with the connection_check approach is that it won't cover the case when the node is responding to `.get_first_point()` method but is not synced. Or there maybe many other failure modes with the node, including:

1. The node is down and not responding.
2. The node is up but not synced with the network.
3. The node is up and is synced with the network but can't handle the load and intermittently returns 500 errors which makes the ETL job fail.

Ideally the solution should cover all of these cases.

What I had in mind is something like this:

```
    def add_export_task(toggle, task_id, python_callable, dependencies=None):
        def python_callable_with_fallback(**kwargs):
            for provider_uri in provider_uris:
                kwargs['provider_uri'] = provider_uri
                try:
                    python_callable(**kwargs)
                except Exception:
                    logging.error('An exception occurred. Trying another uri')

        if toggle:
            operator = python_operator.PythonOperator(
                task_id=task_id,
                python_callable=python_callable_with_fallback,
                provide_context=True,
                execution_timeout=timedelta(hours=15),
                dag=dag,
            )
            if dependencies is not None and len(dependencies) > 0:
                for dependency in dependencies:
                    if dependency is not None:
                        dependency >> operator
            return operator
        else:
            return None
```

Then each of the exporting commands will declare the provider_uri parameter e.g.:

```
def export_blocks_and_transactions_command(execution_date, provider_uri, **kwargs):
```

^ this code is not tested.

Reply:
Implemented the python_callback_with_fallback function and tested on:

1. Disconnection of a node
2. Being out of sync for a node